### PR TITLE
Added new gui.screenMode key to config

### LINKED ISSUE #2319: not able to set the vertical screen mode as default
**Describe the bug**
i want to set the **half screen mode**( achieved by pressing + ) as the default for lazy git. but setting it from the config doesnt seems to be working.

but this mainPanelSplitMode not doing any changes, i tried changing border colors and it changes instantly but not this! 
how do i make the half screen mode by default?
**To Reproduce**
1. set the mainPanelSplitMode on the config file 
```
gui:
  mainPanelSplitMode: 'vertical'  # one of 'horizontal' | 'flexible' | 'vertical'
```

**Expected behavior**
by default lazygit opens up with the vertical screen mode.

**Screenshots**
1. **currrent behaviour when i open lazygit**
current behaviour

2. **expected behaviour when i open lazygit**
expected behavior

**Version info:**
```
version=0.36.0, 
os=linux, 
arch=amd64, 
git version=2.34.1
```

### PR DESCRIPTION
**Pull request by phanithinks**
- **PR Description**
    Fix for #2319 
    new gui.screenMode key added to configuration

- **Please check if the PR fulfills these requirements**

    * [x] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
    * [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
    * [ ] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
    * [x] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
    * [x] Docs (specifically `docs/Config.md`) have been updated if necessary
    * [x] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

**Comment by jesseduffield** [comment]
Looks good @phanithinks . One thing: now that the user is seeing the name 'screenMode', it's worth asking: is this an intuitive name? I think 'windowSize' would be better

**Comment by phanithinks** [comment]
> Looks good @phanithinks . One thing: now that the user is seeing the name 'screenMode', it's worth asking: is this an intuitive name? I think 'windowSize' would be better

Yes, I think thats good name for configuration. I have changed as per your suggestion in configuration.

**Comment by phanithinks** [comment]
@jesseduffield  Do I need to make any more changes to this pr ? Please let me know

File: pkg/config/user_config.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ type GuiConfig struct {
 	CommandLogSize            int                `yaml:"commandLogSize"`
 	SplitDiff                 string             `yaml:"splitDiff"`
 	SkipRewordInEditorWarning bool               `yaml:"skipRewordInEditorWarning"`
+  WindowSize               string              `yaml:"windowSize"`
```
Comment (jesseduffield) [inline pkg/config/user_config.go:54]: This needs to be indented to the right by one more tab

**Review by jesseduffield — CHANGES_REQUESTED** [review]
Sorry for the late re-review, I've left one more comment

File: pkg/config/user_config.go (hunk 1 above)
Comment (phanithinks) [reply pkg/config/user_config.go]: Fixed it. Sorry about that

**Comment by jesseduffield** [comment]
Merged. Thanks for making this @phanithinks !

**Comment by mikeduminy** [comment]
Thank you! Just what I was looking for :D 

One thing to note, the commands for switching between these visual modes refer to this feature as `screenMode`, not `windowSize`. To me `windowSize` feels like it's describing the size of the overall gui window and not the layout. I'd rather go with `gui.layout` or `gui.defaultMainPanelSize`.

**Comment by jesseduffield** [comment]
hmm if there's a time to change that it would be now. I actually think `gui.focusedPanelSize` makes the most sense, as that's the panel that would be resized. It is a shame we just released this haha, meaning we'd have to be backwards compatible

**Pull request merged**